core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+if __name__ == "__main__" and (__package__ is None or __package__ == ''):
+    # replace the script's location in the Python search path by the main
+    # scripts/ folder, above it, so that the importer package folder is in
+    # scope and *not* directly in sys.path; see PEP 395
+    sys.path[0] = str(Path(sys.path[0]).resolve().parent)
+    __package__ = 'importer'
+    # explicitly import the package, which is needed on CPython 3.4 because it
+    # doesn't include https://github.com/python/cpython/pull/2639
+    importlib.import_module(__package__)
+
+from . import cbioportal_common
+from . import libImportOncokb
+from . import validateData
+
+required_cna_columns = ['Hugo_Symbol']
+
+
+# from: cbioportal-frontend file CopyNumberUtils.ts
+cna_alteration_types = {
+    "DELETION": -2,
+    "LOSS": -1,
+    "GAIN": 1,
+    "AMPLIFICATION": 2,
+}
+
+
+def main_import(args):
+
+    study_dir = args.study_directory
+    server_url = args.url_server
+
+    # get a logger to emit messages
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    exit_status_handler = validateData.MaxLevelTrackingHandler()
+    logger.addHandler(exit_status_handler)
+
+    # set default message handler
+    text_handler = logging.StreamHandler(sys.stdout)
+    text_handler.setFormatter(
+        cbioportal_common.LogfileStyleFormatter(study_dir))
+    collapsing_text_handler = cbioportal_common.CollapsingLogMessageHandler(
+        capacity=5e5,
+        flushLevel=logging.CRITICAL,
+        target=text_handler)
+    collapsing_text_handler.setLevel(logging.INFO)
+    logger.addHandler(collapsing_text_handler)
+
+    meta_cna_file_path = libImportOncokb.find_meta_file_by_fields(study_dir, {'genetic_alteration_type': 'COPY_NUMBER_ALTERATION', 'datatype': 'DISCRETE'})
+    cna_file_path = os.path.join(study_dir, libImportOncokb.find_data_file_from_meta_file(meta_cna_file_path))
+    pd_file_name = 'data_cna_pd_annotations.txt'
+    pd_file_path = os.path.join(study_dir, pd_file_name)
+    meta_dict = libImportOncokb.read_meta_file(meta_cna_file_path)
+    if 'pd_annotations_filename' in meta_dict:
+        raise RuntimeError(
+            "Custom driver annotations filename already specified in discrete CNA meta file. Please remove and rerun.")
+    if path.exists(pd_file_path):
+        raise RuntimeError(
+            "Custom driver annotations file '" + pd_file_path + "' for discrete CNA already exists . Please remove and rerun.")
+    check_required_columns(libImportOncokb.get_first_line(open_cna_file(cna_file_path)).rstrip('\n').split('\t'))
+
+    global portal_instance
+    if hasattr(args, 'portal_info_dir') and args.portal_info_dir is not None:
+        portal_instance = validateData.load_portal_info(args.portal_info_dir, logger,
+                                           offline=True)
+    else:
+        portal_instance = validateData.load_portal_info(server_url, logger)
+
+    features = get_features(cna_file_path)
```

Comment:
What does `features` mean here?

Reply:
For the CNA file a feature is a gene-sample-copy number alteration data point.

Reply:
Yeah, that sounds better.

Reply:
Ok, updated.